codec/src/main/java/io/netty/handler/codec/compression/BrotliEncoder.java

```diff
@@ -15,25 +15,50 @@
  */
 package io.netty.handler.codec.compression;
 
+import com.aayushatharva.brotli4j.encoder.BrotliEncoderChannel;
 import com.aayushatharva.brotli4j.encoder.Encoder;
-import com.aayushatharva.brotli4j.encoder.Encoders;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.Unpooled;
 import io.netty.channel.ChannelHandler;
 import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
 import io.netty.handler.codec.MessageToByteEncoder;
 import io.netty.util.ReferenceCountUtil;
 import io.netty.util.internal.ObjectUtil;
 
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.nio.channels.Channels;
+
 /**
- * Compress a {@link ByteBuf} with the brotli format.
- *
+ * Compress a {@link ByteBuf} with the Brotli compression.
+ * <p>
  * See <a href="https://github.com/google/brotli">brotli</a>.
  */
 @ChannelHandler.Sharable
 public final class BrotliEncoder extends MessageToByteEncoder<ByteBuf> {
 
     private final Encoder.Parameters parameters;
+    private PluggableByteBufOutputStream pluggableBuffer;
+    private BrotliEncoderChannel brotliEncoderChannel;
+    private boolean isClosed;
+
+    /**
+     * Encoder flush method is package-private so we have to
+     * use reflection to call that method.
+     */
+    private static final Method FLUSH_METHOD;
+
+    static {
+        Method method;
+        try {
+            method = Encoder.class.getDeclaredMethod("flush");
+            method.setAccessible(true);
```

Comment:
Draft until we can call this directly?

Reply:
Making the API public is scheduled for the next release. For the time being, we will stick to reflection.